doc/microsite/src/main/tut/distage/distage-testkit.md

```diff
@@ -1,14 +1,517 @@
+With these a production test and a dummy test can be introduced for the example
+application. Note how these are the same scenario, `BonusServiceIsZeroTest`,
+but differ in activations.
+
+When extended beyond this small example this pattern simplifies system level tests, sanity checks,
+and even a pragmatic form of
+[N-Version Programming](https://en.wikipedia.org/wiki/N-version_programming)
+
+```scala
+package app
+```
+```scala mdoc:to-string
+final class ProdBonusServiceIsZeroTest extends BonusServiceIsZeroTest with ProdTest
+
+final class DummyBonusServiceIsZeroTest extends BonusServiceIsZeroTest with DummyTest
+```
+
+<pre>
+```scala mdoc:invisible
+// change this block to `passthrough` instead of `invisible` to view test results.
+// The goal is to demonstrate testkit plugin integration. `package` is not
+// currently supported in mdoc code. To hack around this the `package app` code
+// blocks are not interpreted and the actual test tested is the one below.
+izumi.distage.testkit.services.scalatest.dstest.DistageTestsRegistrySingleton.resetRegistry()
+final class MdocBonusServiceIsZeroTest extends BonusServiceIsZeroTest with DummyTest {
+  override def config = super.config.copy(
+    pluginConfig = super.config.pluginConfig.copy(
+      overrides = Seq(BonusServicePlugin)
+    )
+  )
+}
+val t = new MdocBonusServiceIsZeroTest
+org.scalatest.shortstacks.nocolor.run(t)
+```
+</pre>
+
+#### Test Case Context
+
+The `testkit` ScalaTest base classes include the verbs for establishing test context:
+
+- [`should`](https://www.scalatest.org/scaladoc/3.2.0/org/scalatest/verbs/ShouldVerb.html)
+- [`must`](https://www.scalatest.org/scaladoc/3.2.0/org/scalatest/verbs/MustVerb.html)
+- [`can`](https://www.scalatest.org/scaladoc/3.2.0/org/scalatest/verbs/CanVerb.html)
+
+#### Configuration
+
+The test suite class for your application should override the `def config: TestConfig` attributed.
+Among other options, the config defines the plugin configuration, module overrides and activation
+axes. See the d@scaladoc[`TestConfig` API docs](izumi.distage.testkit.TestConfig) for more
+information.
+
+### Syntax Summary
+
+For `F[_]` including `Identity`:
+
+- `in { assert(???) }` - test case is a function returning an
+   [assertion](https://www.scalatest.org/scaladoc/3.2.0/org/scalatest/Assertions.html).
+- `in { (a: A, b: B) => assert(???) }` - test case is a function returning an
+   assertion. The `a` and `b` will be injected from object graph.
+- `in { (a: A, b: B) => ???: F[Unit] }` - test case is a function returning an effect to be
+  executed. The `a` and `b` will be injected from object graph. The test case will fail if the effect
+  fails.
+- `in { (a: A, b: B) => ???: F[Assertion] }` - test case is a function returning an effect to be
+  executed. The `a` and `b` will be injected from object graph. The test case will fail if the effect
+  fails or produces a failure assertion.
+
+For `F[-_, +_, +_]`, same as above with `F[Any, Throwable, _]`:
```

Comment:
@coreyoconnor  Note that using environment does not disable using arguments, the following is valid too:
```
`in { (a: A, b: B) => ???: F[zio.Has[C] with zio.Has[D], Throwable, Assertion] }`
```
All of A, B, C & D will be injected from the object graph